src/pages/Creatures.tsx

```diff
@@ -130,24 +94,14 @@ export default function Creatures() {
                 sx={{ paddingTop: "10px", paddingBottom: "10px" }}
               >
                 <FilterButtons
-                  hasFilters={query.filters.length ? true : false}
-                  setIsFilterDrawerOpen={setIsFilterDrawerOpen}
-                  clearFilters={clearFilters}
+                {...filterButtonProps}
+                />
+                <SearchInput 
+                {...searchInputProps}
                 />
-                <SearchInput q={query.q} qChange={qChange} />
               </TableCell>
             </TableRow>
-            <TableRow role="presentation">
-              <TablePagination
-                role="presentation"
-                count={count}
-                page={query.page}
-                labelRowsPerPage="Num: "
-                onPageChange={pageChange}
-                rowsPerPage={query.size}
-                onRowsPerPageChange={sizeChange}
-              />
-            </TableRow>
+            <TableRow role="presentation">{InstanceTablePagination}</TableRow>
```

Comment:
Lets still pass the specific props here so we don't couple the hook to the component too much.

You can make code so DRY it becomes brittle.